### ISSUE TITLE
Use language of song/album/library when using link

### ISSUE THREAD

**Issue by StayBlue** [lyarenei/jellyfin-plugin-itunes#8]
By default, the plugin uses music.apple.com/us. My proposal involves checking the current language of the song/album/library (in that order of precedence) when driving the metadata instead. This allows the metadata to be based on the language instead of always being in English.

**Comment by lyarenei** [comment]
@StayBlue This should be now implemented in #9, if you want, you can test it by either downloading the build from the CI jobs or you can temporarily switch to my dev repo (https://repo.xkrivo.net/jellyfin-dev/manifest.json) and install plugin version `2.1.0.0`

**Comment by StayBlue** [comment]
Couldn't the "Country/Region" field in the library be used? Sometimes the songs aren't tagged properly.

**Comment by lyarenei** [comment]
@StayBlue As far as I know, the field is only for metadata language settings and has nothing to do with `Language` tag in the song file metadata. Or at least I have never seen this field set on my server despite having all my music tagged with Picard. Except when I set it explicitly for testing these changes.

Otherwise the priority should be hierarchical, with song level having the highest priority and server global settings having the least priority. So as long as the region field is not set on lower levels than the library, it should behave as you want/describe.

**Comment by StayBlue** [comment]
I am using the plugin in the development repository, and pressing the button that allows you to go to the link that the album ID is set to always redirects to: [https://music.apple.com/us/album/0](https://music.apple.com/us/album/0).

**Comment by Druidblack** [comment]
confirm. Version 2.1.0.0 does not work. For previously received information (from the old version of the plugin) Version 2.1.0.0 shows the link https://music.apple.com/us/album/0
I returned to version 2.0.0.1. The links started working fine

**Comment by lyarenei** [comment]
Hmm, for some reason, I can't reproduce it locally, so I have never noticed, sorry. But it was broken on my server as well and I found the issue. It should be fixed in version `2.1.0.1`.

---

Btw have I just found out that I have never released v2 to my main repo? Uh, my head hurts... 🤦

**Comment by Druidblack** [comment]
Version 2.0.0.0 takes into account the changes from 2.1.0.1. ?

**Comment by lyarenei** [comment]
> Version 2.0.0.0 takes into account the changes from 2.1.0.1. ?

No, I noticed that 2.0.0.0 is not in the main repo and does not have a tagged release in this repository. It just looked that I have never updated the plugin to work with 10.9.x, but in reality, I did.


> Based on which setting in jellyfin will the plugin determine which language to use for uploading data?

As I mentioned in earlier comment, the metadata language for Apple Music is decided by the region field, not the laguage field in metadata.

**Comment by StayBlue** [comment]
> Hmm, for some reason, I can't reproduce it locally, so I have never noticed, sorry. But it was broken on my server as well and I found the issue. It should be fixed in version `2.1.0.1`.
> 
> Btw have I just found out that I have never released v2 to my main repo? Uh, my head hurts... 🤦

The issue has been fixed in the latest version, but only somewhat. The link redirects to the US link of Apple Music solely, despite my library settings being set to a different country/region and language.

**Comment by Druidblack** [comment]
Version 2.1.0.1 does not work.

With version 2.1.0.1, when updating metadata for albums and artists, a link to Apple Music is not added.

Rolled back to version 2.0.0.1, checked. On this version, when updating metadata, a link to Apple Music is added automatically.

Version 2.1.0.0. I can't check because of an error in the link.

P.S. Having studied the work of the plug-in and the Apple Music service, I want to share my idea. For music albums, you can leave a us region link. That's how the name of the Apple Music albums does not change to a transcription.
We need to solve the problem with the album artists. Since Apple Music uses the us region, it sends the names of the artists in transcription (this breaks down the work of matching songs with the artist in jf). If the plugin can get information about the artist based on the selected region in the library, it will be good.

**Comment by Druidblack** [comment]
Additionally, I checked the recognition of performers.
Version 2.1.0.1 Could not find the artist. There were no Apple Music options in the selection window. The search was conducted in the language of the region (specified in the library settings) 
Version 2.0.0.1 Under the same search conditions. I immediately displayed it in a window with options for the desired author.

**Comment by Druidblack** [comment]
I installed the 2.1.0.1 plugin on a clean system and added a new music library to it.

For artists, the Apple Music id is not added to the metadata during the new scan (as it was in version 2.0.0.1)
But at the same time, when searching for images of artists, the plugin searches by the artist's name.

![1](https://github.com/user-attachments/assets/61cd9590-b401-4dd9-aaa9-92ca0b821040)
![2](https://github.com/user-attachments/assets/7093944e-4a3c-4a75-a886-df61e0921847)

**Comment by Druidblack** [comment]
I installed the 2.1.0.1 plugin on a clean system and added a new music library to it.

No Apple Music id is assigned for albums. Album art is apparently searched only by the album title. Because of this, he does not find the covers or finds the wrong ones.
![5](https://github.com/user-attachments/assets/dcf56ee5-a999-461c-a354-0faf586de4ac)
![4](https://github.com/user-attachments/assets/a8fca7b2-b40b-4048-a5f5-d88b9f97cf90)

**Comment by Druidblack** [comment]
This is how version 2.0.0.1 works. It works when updating (adding) albums and artists (I have 99% identification accuracy, only one artist was identified incorrectly)
For artists and albums, an Apple Music id is assigned and image search is already performed using the id. Due to the disadvantages of the US region, names other than English are loaded into transcriptions.
![8](https://github.com/user-attachments/assets/15dab2ca-69a9-4ec5-b491-92e0710949fe)
![7](https://github.com/user-attachments/assets/a231a08b-9c29-4bdd-bc57-009d860ee37d)
![6](https://github.com/user-attachments/assets/dfb50b16-1d6f-4a18-ab6f-324bbf4e6110)
![9](https://github.com/user-attachments/assets/075d50e8-5cb9-4105-924d-fcaf5bb38c5f)

**Comment by lyarenei** [comment]
> The issue has been fixed in the latest version, but only somewhat. The link redirects to the US link of Apple Music solely, despite my library settings being set to a different country/region and language.

@StayBlue I see, this is actually not possible. I had hoped that maybe the URLs were saved as well, but it's actually only the IDs. So the only variable which is expected in the URL is the ID itself:

https://github.com/jellyfin/jellyfin/blob/0bc142de9acd01b8b9191b8e99da41fdebabe0a7/MediaBrowser.Providers/Manager/ProviderManager.cs#L908-L911

There are two ways I can think of how to overcome this. 
- The first and proper solution is to modify the `ExternalId` interface in Jellyfin. Depending on the approach, this could be relatively easy (in terms of the complexity). 
- The other solution is to add a plugin setting which will set the language globally for the whole plugin - but this would modify language in all links, regardless of the  language set in metadata/library (except when searching).

---

As for the search results - if the region is set to US, then there should be no differences in results between versions `2.1.x` and `2.0.x`. I think it's possible to have artist/album on Apple Music show completely fine in the US region and not in let's say Germany (DE). This can be verified by doing search manually (this is what the plugin does after all). Also more details can be seen in the logs with debug logging enabled.
